### PR TITLE
mehmet bozdemir nodeJS week3 homework

### PR DESCRIPTION
**Pull request by mehmet-bozdemir**

### CODE REVIEW

**Comment by mehmet-bozdemir** [comment]
Dear Andrej, for some reason I skipped the main index.js file  so I made a new commit. Now you can check it but I couldn't figure out why all the files in week3/homework/src directory are apart from each other.

File: week3/homework/src/index.js [hunk 1]
```diff
@@ -1,3 +1,132 @@
...
+  const todo = todos.find(todo => todo.id == req.params.id);
+  // console.log(todo);
+  todo.isDone = false;
+  await saveTodos(todos);
+  res.status = 201;
+  res.end();
+}
+
+async function clearTodos(req, res) {
+  await saveTodos([]);
```
Comment (gajduk) [inline week3/homework/src/index.js]: nice and simple 👍

File: week3/homework/src/index.js [hunk 2]
```diff
@@ -1,3 +1,132 @@
...
+}
+
+app.get('/todos', (req, res) => readTodos(req, res));
+app.post('/todos', (req, res) => createTodo(req, res));
+app.put('/todos/:id', (req, res) => updateTodo(req, res));
+app.delete('/todos/:id', (req, res) => deleteTodo(req, res));
+
+// homework
+
+app.get('/todos/:id', (req, res) => readTodo(req, res));
```
Comment (gajduk) [inline week3/homework/src/index.js]: very clean 👍

File: week3/homework/src/index.js [hunk 3]
```diff
@@ -1,3 +1,132 @@
...
+  let todos = await readTodosFromFile();
+  let todo = todos.find(todo => todo.id == req.params.id);
+  // console.log(todo);
+  res.send(todo);
+}
+
+async function markAsDone(req, res) {
+  const todos = await readTodosFromFile();
+  const todo = todos.find(todo => todo.id == req.params.id);
+  todo.isDone = true;
```
Comment (gajduk) [inline week3/homework/src/index.js]: what would happen if there is no todo with the given id?

**Comment by mehmet-bozdemir** [comment]
Dear Andrej, thanks for your valuable feedback. Every day I am learning new stuff and building knowledge,  for the next step I am gonna consider perfectness as you mentioned in your feedback, handling errors etc.  We are learning a lot from you.

**Pull request closed without merging**